float32[4] row1
float32[4] row2
float32[4] row3
float32[4] row4